core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -329,6 +338,38 @@ public synchronized TopLevelItem createProject( TopLevelItemDescriptor type, Str
     }
 
     /**
+     * Populate a {@link Categories} object from a specific {@link ItemGroup}.
+     *
+     * @return A object that represents a set of {@link Category}
+     */
+    public static Categories getCategories(Authentication a, ItemGroup c) {
+        Categories categories = new Categories();
+        int weight = ItemCategory.MIN_WEIGHT;
+        for (TopLevelItemDescriptor descriptor : DescriptorVisibilityFilter.apply(c, Items.all(a, c))) {
```

Comment:
:ant: Possibly you need to be passing a `View`, not an `ItemGroup`, as the `context`; you are calling this from `View` anyway, so you have that context. The reason is that `View/newJob.jelly` has always [used that as the `context`](https://github.com/jenkinsci/jenkins/blob/a5124b5fd504291c7d0c7dcbadd84874d1486952/core/src/main/resources/lib/hudson/newFromList/form.jelly#L53) and some implementations might be relying on that detail, for example to restrict items added to a particular view, though so far the examples I have found will accept _either_ `View` _or_ one of `ItemGroup` or `ViewGroup` (`Jenkins` and `Folder` implement both).